lms/djangoapps/gating/api.py

```diff
@@ -63,35 +61,60 @@ def _get_subsection_percentage(subsection_grade):
     """
     Returns the percentage value of the given subsection_grade.
     """
-    if subsection_grade.graded_total.possible:
-        return float(subsection_grade.graded_total.earned) / float(subsection_grade.graded_total.possible) * 100.0
-    else:
-        return 0
+    return _calculate_percentage(subsection_grade.graded_total.earned, subsection_grade.graded_total.possible) * 100.0
 
 
-def evaluate_entrance_exam(course, subsection_grade, user):
+def _calculate_percentage(earned, possible):
+    """
+    Returns the percentage of the given earned and possible values.
+    """
+    return float(earned) / float(possible) if possible else 0.0
+
+
+def evaluate_entrance_exam(course_grade, user):
     """
     Evaluates any entrance exam milestone relationships attached
-    to the given subsection. If the subsection_grade meets the
-    minimum score required, the dependent milestone will be marked
+    to the given course. If the course_grade meets the
+    minimum score required, the dependent milestones will be marked
     fulfilled for the user.
     """
+    course = course_grade.course
     if milestones_helpers.is_entrance_exams_enabled() and getattr(course, 'entrance_exam_enabled', False):
-        subsection = modulestore().get_item(subsection_grade.location)
-        in_entrance_exam = getattr(subsection, 'in_entrance_exam', False)
-        if in_entrance_exam:
-            # We don't have access to the true request object in this context, but we can use a mock
-            request = RequestFactory().request()
-            request.user = user
-            exam_pct = get_entrance_exam_score(request, course)
-            if exam_pct >= course.entrance_exam_minimum_score_pct:
-                exam_key = UsageKey.from_string(course.entrance_exam_id)
+        if get_entrance_exam_content(user, course):
+            exam_chapter_key = get_entrance_exam_usage_key(course)
+            exam_percentage = get_entrance_exam_percentage(course_grade, exam_chapter_key)
+            if exam_percentage >= course.entrance_exam_minimum_score_pct:
```

Comment:
It just seems wild to me that this line is comparing `0.6345` against `0.8000`, instead of `63.45` against `80`.